### PR TITLE
Improve DTO mapping and pet creation

### PR DESCRIPTION
**Pull request by misto**
This pull request improves the mapping between pets/visits DTOs to their entities. When adding a pet, HTTP status CREATED and the new location is returned (similar to the other controllers).

### CODE REVIEW

File: src/main/java/org/springframework/samples/petclinic/rest/controller/PetRestController.java [hunk 1]
```diff
@@ -98,7 +100,10 @@ public ResponseEntity<PetDto> deletePet(Integer petId) {
     @PreAuthorize("hasRole(@roles.OWNER_ADMIN)")
     @Override
     public ResponseEntity<PetDto> addPet(PetDto petDto) {
-        this.clinicService.savePet(petMapper.toPet(petDto));
-        return new ResponseEntity<>(petDto, HttpStatus.OK);
+        HttpHeaders headers = new HttpHeaders();
+        Pet pet = petMapper.toPet(petDto);
+        this.clinicService.savePet(pet);
+        headers.setLocation(UriComponentsBuilder.newInstance().path("/api/pets/{id}").buildAndExpand(pet.getId()).toUri());
+        return new ResponseEntity<>(petMapper.toPetDto(pet), headers, HttpStatus.CREATED);
```
[the remark this answers is not in the document]
Comment (misto) [reply src/main/java/org/springframework/samples/petclinic/rest/controller/PetRestController.java:107]: Oh yes, I missed that. I've added another commit.

File: src/main/java/org/springframework/samples/petclinic/mapper/PetMapper.java [hunk 2]
```diff
@@ -23,6 +23,7 @@ public interface PetMapper {
 
     Collection<Pet> toPets(Collection<PetDto> pets);
 
+    @Mapping(source = "ownerId", target = "owner.id")
```
[the remark this answers is not in the document]
Comment (misto) [reply src/main/java/org/springframework/samples/petclinic/mapper/PetMapper.java:26]: I noticed that the mapping was missing when I wrote a JMeter test that creates new pets with an owner (for this blog post: https://medium.com/growing-green-software/evolution-of-energy-usage-in-spring-boot-69c7c372dba3). I guess I could have used the OwnerRestController::addPetToOwner instead, but fixing it like this seemed like a good idea :-)

[the remark this answers is not in the document]
Comment (arey) [reply src/main/java/org/springframework/samples/petclinic/mapper/PetMapper.java:26]: It's a very interesting article. Thanks. I just references it : https://spring-petclinic.github.io/docs/resources.html 
I didn't know the `JoularJX` tool. I'm going to consider it in my GreenIT deployment work

[the remark this answers is not in the document]
Comment (misto) [reply src/main/java/org/springframework/samples/petclinic/mapper/PetMapper.java:26]: Thanks 😊 Let me know if you have any ideas about what could be benchmarked with the Petclinic (I already did JVM versions and embedded web servers; data access might be another one to try) or if you have other applications.

[the remark this answers is not in the document]
Comment (arey) [reply src/main/java/org/springframework/samples/petclinic/mapper/PetMapper.java:26]: Hi @misto what do you think about comparing the Java and the Kotlin version? I don't think there are any difference. But who knows?

**Pull request merged**